framework/src/problems/FEProblemBase.C

```diff
@@ -5859,6 +5856,8 @@ FEProblemBase::computeResidualAndJacobian(const NumericVector<Number> & soln,
 
     for (const auto & residual_vector_tag : residual_vector_tags)
       _fe_vector_tags.insert(residual_vector_tag._id);
+
+    setCurrentResidualVectorTags(_fe_vector_tags);
```

Comment:
I'm a little confused on why this needs to be FEProblemBase and can't be pushed down to NonlinearSystemBase. Is it because `setCurrentResidualVectorTags` is telling what vectors to accumulate and the `tags` function parameter is telling what residual objects to go through?

Reply:
To me it's about data locality. In almost all cases the tag computation begins with `_fe_problem.computeResidualTags`. And then when it comes time to accumulate residuals into the global data structures it happens through  `_fe_problem.XXXResidual`. Any time I call `_fe_problem.XXXResidual` the problem would have to query the nonlinear system base object for the residual vector tags; I just don't see the reason for the indirection

Reply:
Additionally, the problem objects are where are all the other vector tag data is stored, e.g. `_vector_tags`, `_typed_vector_tags`, etc.

Reply:
Ahh, that makes more sense. I just thought it strange that CrankNicolson had to call this function too.

Reply:
What's unfortunate is that CrankNicolson has to call the `nl.computeResidualTag` function directly. This direct call also happens in some Griffin solve objects. I would have liked to make that function private if I could have